### ISSUE TITLE
Syntax warning in lib/python3.9/site-packages/tensorflow/python/keras/benchmarks/benchmark_util.py

### ISSUE THREAD

**Issue by Nizarazo** [tensorflow/tensorflow#49910]
Hi

After moving to python3.9 I get syntax warning in: 
lib/python3.9/site-packages/tensorflow/python/keras/benchmarks/benchmark_util.py lines 142, 144, 146

error message:
pythonVirtualEnv/lib/python3.9/site-packages/tensorflow/python/keras/benchmarks/benchmark_util.py:142: SyntaxWarning: "is" with a literal. Did you mean "=="?

Code causing error:
if 'x' is None:
    raise ValueError('Input data is required.')
  if 'optimizer' is None:
    raise ValueError('Optimizer is required.')
  if 'loss' is None:
    raise ValueError('Loss function is required.')

'x' 'optimizer' 'loss' isn't None since it is a string so python3.9 do not like it and raise a warning.
In order to fix it I removed those lines locally then no longer getting syntax warning.

Can you fix it? And when?

Thanks

**Comment by tilakrayal** [comment]
@Nizarazo ,

In order to expedite the trouble-shooting process, could you please provide a minimal code snippet and the TensorFlow version you are using.

Thanks!

**Comment by Nizarazo** [comment]
Hi @tilakrayal 

Thanks for your reply.

I am using tensorflow 2.5.0 release, the code snippet is in my previous comment I will post it again here:
f 'x' is None:
raise ValueError('Input data is required.')
if 'optimizer' is None:
raise ValueError('Optimizer is required.')
if 'loss' is None:
raise ValueError('Loss function is required.')

This code is located in tensorflow/python/keras/benchmarks/benchmark_util.py line 142

Anything else needed?

**Comment by Nizarazo** [comment]
Hi @tilakrayal

Will it be fixed? 
if yes when and in which version?

Thanks

**Comment by tilakrayal** [comment]
@Nizarazo ,

I was able to execute the above mentioned code without any errors.Please find the gist [here](https://colab.research.google.com/gist/tilakrayal/57733590354f2fd55ccb155ce310dd7e/49910.ipynb).

Can you please provide gist which you are facing issue in executing the code.It helps to analyse the issue.

Thanks!

**Comment by Nizarazo** [comment]
Hi

We build exe using pyinstaller and one of our packages is tensorflow so while building the exe using the pyinstaller raises an error due to this code located in tensorflow/keras and as a result it stops building the executable.
So here is the errors from pyinstaller when it analysis your tensorflow/keras module:

Showing All Messages
pythonVirtualEnv/lib/python3.9/site-packages/tensorflow/python/keras/benchmarks/benchmark_util.py:142: SyntaxWarning: "is" with a literal. Did you mean "=="?

  if 'x' is None:

pythonVirtualEnv/lib/python3.9/site-packages/tensorflow/python/keras/benchmarks/benchmark_util.py:144: SyntaxWarning: "is" with a literal. Did you mean "=="?

  if 'optimizer' is None:

pythonVirtualEnv/lib/python3.9/site-packages/tensorflow/python/keras/benchmarks/benchmark_util.py:146: SyntaxWarning: "is" with a literal. Did you mean "=="?

  if 'loss' is None:

You can reproduce the issue if you run pyinstaller with tensorflow as a requirement in your python environment.

Thanks

**Comment by Nizarazo** [comment]
Hi @Saduf2019 

Any more info is needed or we are all set and waiting for your fix?

Thanks

**Comment by Nizarazo** [comment]
Hi @Saduf2019 

I am still waiting your reply regarding my previous comment, is there any more info needed?

Thanks

**Comment by Saduf2019** [comment]
@Nizarazo 
Would you want to try with python 3.8 as there are no errors with it.

**Comment by Nizarazo** [comment]
Hi @Saduf2019 

No because python3.8 do not support Big Sur it needs to be supported in python3.9 since it is the first python to support Big Sur 11 please see documentation here:
https://www.python.org/downloads/release/python-391/

Thanks

**Comment by Nizarazo** [comment]
Hi @jvishnuvardhan 

Is there any more info needed for this ticket or we are all set and waiting for the fix?

Thanks

**Comment by Nizarazo** [comment]
Hi @jvishnuvardhan 

Any news regarding the fix?

Thanks

**Comment by Nizarazo** [comment]
Hi @jvishnuvardhan 

What does it mean "awaiting tensorflower" status?

Thanks

**Comment by jvishnuvardhan** [comment]
@Nizarazo It means it was assigned to a Tensorflow code owner and waiting for the code owner to respond once root-cause was found. Thanks!

**Comment by Nizarazo** [comment]
Hi @jvishnuvardhan 

So when the code owner supposed to respond on that it has been in "awaiting tensorflower" state for for 7 days?

Thanks

**Comment by Nizarazo** [comment]
Hi @jvishnuvardhan 

Still waiting for your reply regarding my previous comment.

Thanks

**Comment by Nizarazo** [comment]
Hi @jvishnuvardhan 

I am still waiting for the code owner to fix it any idea what is the status of it? 
I have been waiting more than one month already since 16.6.

Thanks

**Comment by sachinprasadhs** [comment]
Hi, Could you try again with the latest stable Tensorflow version 2.9.1 with the support of Python 3.10 and let us know if you still face an issue. Thanks!
